provider/core/src/resource.rs

```diff
@@ -25,12 +25,28 @@ impl ResourceKeyHash {
     }
 }
 
-/// The resource key used for loading data from an ICU4X data provider.
+/// Used for loading data from an ICU4X data provider.
 ///
 /// A resource key is tightly coupled with the code that uses it to load data at runtime.
-/// Executables can be searched for ResourceKey instances to produce optimized data files.
+/// Executables can be searched for `ResourceKey` instances to produce optimized data files.
 /// Therefore, users should not generally create ResourceKey instances; they should instead use
 /// the ones exported by a component.
+///
+/// `ResourceKey`s are created with the [`resource_key!`] macro:
+///
+/// ```
+/// # use icu_provider::prelude::ResourceKey;
+/// const key: ResourceKey = icu_provider::resource_key!("foo/bar@1");
+/// ```
+///
+/// The path string has to contain at least one `/`, and end with `@` followed by one or more ASCII
+/// digits. Paths do not contain characters other than ASCII letters and digits, `_`, `/`, `=`, and
+/// `@`. Invalid paths are compile-time errors (as [`resource_key!`] is `const`).
```

Comment:
```suggestion
/// `@`. Invalid paths are compile-time errors (as [`resource_key!`] uses `const`).
```

nit: macros don't have a notion of constness